src/coprocessor/codec/table.rs

```diff
@@ -183,7 +183,8 @@ fn unflatten(ctx: &EvalContext, datum: Datum, col: &ColumnInfo) -> Result<Datum>
         types::LONG_BLOB |
         types::VARCHAR |
         types::STRING |
-        types::NEW_DECIMAL => Ok(datum),
+        types::NEW_DECIMAL |
```

Comment:
I think this can be merged with L197~L200 using `debug_assert`. An error log can do nothing until it's release since we don't see log when running tests (unless it fails).